packages/perseus-editor/src/widgets/video-editor/video-editor.tsx

```diff
@@ -0,0 +1,33 @@
+import {EditorJsonify} from "@khanacademy/perseus";
+import {
+    videoLogic,
+    type VideoDefaultWidgetOptions,
+    type PerseusVideoWidgetOptions,
+} from "@khanacademy/perseus-core";
+import * as React from "react";
+
+import VideoSettings from "./video-settings";
+
+export interface VideoEditorProps extends PerseusVideoWidgetOptions {
+    onChange: (options: PerseusVideoWidgetOptions) => void;
+}
+
+/**
+ * This is the main editor for this widget, to specify all the options.
+ */
+class VideoEditor extends React.Component<VideoEditorProps> {
+    static widgetName = "video" as const;
+
+    static defaultProps: VideoDefaultWidgetOptions =
+        videoLogic.defaultWidgetOptions;
+
+    serialize: () => any = () => {
+        return EditorJsonify.serialize.call(this);
+    };
```

Comment:
I'd really like to move away from using this `EditorJsonify` helper. It's a very "fast and loose" way of serializing widget options. 

Instead, these `serialize` functions should be marked as returning the type of widget options for the widget they edit. So in this case, I'm pretty sure this editor should serialize out [`PerseusVideoWidgetOptions`](https://github.com/Khan/perseus/blob/42c456d99e0002589c088c721f09cbb26ea57b2c/packages/perseus-core/src/data-schema.ts#L1701) objects and this function is as simple as:


```suggestion
    serialize: () => PerseusVideoWidgetOptions = () => {
        return {
            location: this.props.location,
            static: this.props.static,
        }
    };
```